packages/adapters/src/process-sequence-adapter.ts

```diff
@@ -0,0 +1,109 @@
+import { SequenceInfo } from "./sequence-info";
+import { isValidSequencePackageJSON } from "./validate-sequence-package-json";
+
+const HOME_DIR = require("os").homedir();
+const SEQUENCES_DIR = path.join(HOME_DIR, ".scramjet_sequences");
+
+async function getRunnerConfigForStoredSequence(id: string): Promise<ProcessSequenceConfig> {
+    const packageJsonPath = path.join(SEQUENCES_DIR, id, "package.json");
+    const packageJson = await readStreamedJSON(createReadStream(packageJsonPath));
+
+    if (!isValidSequencePackageJSON(packageJson)) {
+        throw new Error("Invalid Scramjet sequence package.json");
+    }
+
+    return {
+        type: "process",
+        sequencePath: packageJson.main,
+        version: packageJson.version ?? "",
+        name: packageJson.name ?? "",
+        id
+    };
+}
+
+export function getSequenceDir(id: string): string {
+    return path.join(SEQUENCES_DIR, id);
+}
+
+class ProcessSequenceAdapter implements ISequenceAdapter {
+    private logger: Logger;
+    private computedInfo: ISequenceInfo | null = null
+
+    constructor(info?: ISequenceInfo) {
+        if (info && info.getConfig().type !== "process") {
+            throw new Error("Invalid info config for ProcessSequenceAdapter");
+        }
+
+        this.computedInfo = info ?? null;
+        this.logger = getLogger(this);
+    }
+
+    public get info(): ISequenceInfo {
+        if (!this.computedInfo) {
+            throw new Error("Sequence not identified yet");
+        }
+
+        return this.computedInfo;
+    }
+
+    async init(): Promise<void> {
+        if (!existsSync(SEQUENCES_DIR)) {
+            await mkdir(SEQUENCES_DIR);
+        }
+    }
+
+    async list(): Promise<ProcessSequenceAdapter[]> {
+        const storedSequencesIds = await readdir(SEQUENCES_DIR);
+        const sequencesConfigs = await Promise.all(
+            storedSequencesIds
+                .map(getRunnerConfigForStoredSequence)
+                .map((configPromised) => configPromised.catch(() => null))
+        );
+
+        this.logger.debug("Listed stored sequences", sequencesConfigs);
+
+        return sequencesConfigs
+            .filter(isDefined)
+            .map((config) => new SequenceInfo(config))
+            .map((info) => new ProcessSequenceAdapter(info));
+    }
+
+    async identify(stream: Readable, id: string): Promise<void> {
+        const sequenceDir = getSequenceDir(id);
+
+        await mkdir(sequenceDir);
+
+        const uncompressingProc = exec(`tar zxf - -C ${sequenceDir}`);
+
+        stream.pipe(uncompressingProc.stdin!);
+
+        await new Promise(res => uncompressingProc.on("close", res));
+
+        const config = await getRunnerConfigForStoredSequence(id);
+
+        this.computedInfo = new SequenceInfo(config);
+    }
+
+    async cleanup(): Promise<void> {
+        return rm(getSequenceDir(this.info.getId()), { recursive: true });
+    }
+
+    async remove() {
+        //noop
+    }
```

Comment:
there was a bit of ambiguity between `remove` and `cleanup`. So I removed `cleanup` xd

Reply:
Hmm... Cleanup was supposed to remove any adapter level metadata in the lifecycle. not to remove the specific sequence info. This may come in handy again.